src/app/GameController.ts

```diff
@@ -0,0 +1,45 @@
+import { AnimalNames } from '../Enums/AnimalNamesEnum';
+import { Player } from '../Player';
+import { BreedProcessor } from './BreedProcessor';
+import { Timer } from './Timer';
+
+export class GameController {
+  private currentPlayer: Player;
+  private player: Player;
+  private timer: Timer;
+  private breedProcessor: BreedProcessor;
+  constructor(playerName: string) {
+    this.player = new Player(playerName);
```

Comment:
So when a game controller is invoked, a new player is created? Perhaps it would be good to create a player as a separete method (initializeNewPlayer or something like that?). Properties of Game Controller player and currentPlayer are confusing.

Reply:
+1
Also player can pick avatar he wants to use. it should be added in players constrctor

Reply:
It'll work differently with more players, so I'll write down that suggestion to consider while working on multiplayer solution. Current player is also prepared for multiplayer game, then those names shouldn't be that confusing anymore.